public/components/metrics/redux/slices/metrics_slice.ts

```diff
@@ -87,22 +94,31 @@ export const loadMetrics = () => async (dispatch) => {
   const customDataRequest = fetchCustomMetrics();
   const remoteDataSourcesResponse = await pplServiceRequestor(pplService!, PPL_DATASOURCES_REQUEST);
   const remoteDataSources = remoteDataSourcesResponse.data.DATASOURCE_NAME;
-
   dispatch(setDataSources(remoteDataSources));
   dispatch(setDataSourceTitles(remoteDataSources));
   dispatch(
-    setDataSourceIcons(coloredIconsFrom([OBSERVABILITY_CUSTOM_METRIC, ...remoteDataSources]))
+    setDataSourceIcons(
+      coloredIconsFrom([OBSERVABILITY_CUSTOM_METRIC, ...remoteDataSources, 'OpenTelemetry'])
+    )
   );
 
   const remoteDataRequests = await fetchRemoteMetrics(remoteDataSources);
   const metricsResultSet = await Promise.all([customDataRequest, ...remoteDataRequests]);
+  // console.log('metricsResultSet: ', metricsResultSet);
   const metricsResult = metricsResultSet.flat();
-
+  // console.log('metricsResult: ', metricsResult);
   const metricsMapById = keyBy(metricsResult.flat(), 'id');
   dispatch(mergeMetrics(metricsMapById));
 
   const sortedIds = sortBy(metricsResult, 'catalog', 'id').map((m) => m.id);
-  dispatch(setSortedIds(sortedIds));
+  // console.log('sortedIds: ', sortedIds);
+  await dispatch(setSortedIds(sortedIds));
+  // await dispatch(setOtelIndices(fetchOTindices));
+};
+
+export const loadOTIndices = () => async (dispatch) => {
```

Comment:
why are these functions returning functions?

Reply:
changed it